cloud/user/user.go

```diff
@@ -225,3 +240,227 @@ func ListOrgUsers(out io.Writer, client astrocore.CoreClient) error {
+		return err
+	}
+	fmt.Fprintf(out, "The workspace user %s role was successfully updated to %s\n", email, role)
+	return nil
+}
+
+// IsWorkspaceRoleValid checks if the requested role is valid
+// If the role is valid, it returns nil
+// error ErrInvalidWorkspaceRole is returned if the role is not valid
+func IsWorkspaceRoleValid(role string) error {
+	validRoles := []string{"WORKSPACE_MEMBER", "WORKSPACE_BILLING_ADMIN", "WORKSPACE_OWNER"}
+	for _, validRole := range validRoles {
+		if role == validRole {
+			return nil
+		}
+	}
+	return ErrInvalidWorkspaceRole
+}
+
+// Returns a list of all of an organizations users
+func GetWorkspaceUsers(client astrocore.CoreClient, workspace string, limit int) ([]astrocore.User, error) {
+	offset := 0
+	var users []astrocore.User
+
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return nil, err
+	}
+	if ctx.OrganizationShortName == "" {
+		return nil, ErrNoShortName
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+	for {
+		resp, err := client.ListWorkspaceUsersWithResponse(httpContext.Background(), ctx.OrganizationShortName, workspace, &astrocore.ListWorkspaceUsersParams{
+			Offset: &offset,
+			Limit:  &limit,
+		})
+		if err != nil {
+			return nil, err
+		}
+		err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		users = append(users, resp.JSON200.Users...)
+
+		if resp.JSON200.TotalCount <= offset {
+			break
+		}
+
+		offset += limit
+	}
+
+	return users, nil
+}
+
+// Prints a list of all of an organizations users
+func ListWorkspaceUsers(out io.Writer, client astrocore.CoreClient, workspace string) error {
+	table := printutil.Table{
+		Padding:        []int{30, 50, 10, 50, 10, 10, 10},
+		DynamicPadding: true,
+		Header:         []string{"FULLNAME", "EMAIL", "ID", "WORKSPACE ROLE", "CREATE DATE"},
+	}
+	users, err := GetWorkspaceUsers(client, workspace, userPagnationLimit)
+	if err != nil {
+		return err
+	}
+
+	for i := range users {
+		table.AddRow([]string{
+			users[i].FullName,
+			users[i].Username,
+			users[i].Id,
+			*users[i].WorkspaceRole,
+			users[i].CreatedAt.Format(time.RFC3339),
+		}, false)
+	}
+
+	table.Print(out)
+	return nil
+}
+
+func DeleteWorkspaceUser(email, workspace string, out io.Writer, client astrocore.CoreClient) error {
+	var userID string
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return ErrNoShortName
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+	// Get all org users. Setting limit to 1000 for now
+	users, err := GetWorkspaceUsers(client, workspace, userPagnationLimit)
+	if err != nil {
+		return err
+	}
+	if email == "" {
+		user, err := selectUser(users, true)
+		userID = user.Id
+		email = user.Username
+		if err != nil {
+			return err
+		}
+	} else {
+		for i := range users {
+			if users[i].Username == email {
+				userID = users[i].Id
+			}
+		}
+	}
```

Comment:
Can we refactor this into its own method as it is duplicated in all three `AddWorkspaceUser`,  `UpdateWorkspaceUserRole` & `DeleteWorkspaceUser`?

Reply:
That makes sense